util/fibers/proactor_base.cc

```diff
@@ -234,6 +240,17 @@ void ProactorBase::RegisterSignal(std::initializer_list<uint16_t> l, std::functi
   }
 }
 
+void ProactorBase::ProcessSleepFibers(detail::Scheduler* scheduler) {
+  // avoid calling steady_clock::now() too much.
+  // Cycles count can reset, for example when CPU is suspended, therefore we also allow
+  // "returning  into past". False positive is possible but it's not a big deal.
+  uint64_t now = GetCPUCycleCount();
+  if (now < last_sleep_cycle_ || now >= last_sleep_cycle_ + cycles_per_10us) {
```

Comment:
Could you add a warning about the time jump?

Reply:
I can - but FYI, this does not affect correctness of this function since `ProcessSleep` uses a real clock anyways.

Reply:
Please tell me if you still find this warning useful.

Reply:
Yeah, seems to me like a thing that might be useful one day and it's rare enough that I don't care about log polution :)